unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NotificationMessagesHUD/ChatNotificationController.cs

```diff
@@ -137,53 +137,57 @@ private void HandleMessageAdded(ChatMessage[] messages)
                 if (notificationEntries.Contains(message.messageId)) return;
                 notificationEntries.Add(message.messageId);
 
-                AddNotification(message, channel).Forget();
+                AddNotificationAsync(message, channel).Forget();
```

Comment:
Add cancellation token at least for the Dispose flow if we allow multiple instances of this task running at the same time.

Reply:
The only async operation here is the profanity filtering, but it does not support cancellation tokens, and gets out of the scope of this pr to add it. I can add a TODO comment though

Reply:
That's fine, you can pass the CT to the profanity filtering (you can add it as `CancellationToken ct = default` to avoid changing signatures across the project) and ignore it. But at least this layer would have proper disposal. Then in another ticket we can add cancellation to the profanity filter.

Reply:
Ok!